src/VisualStudio/Core/Def/Implementation/ColorSchemes/ColorSchemeApplier.ForegroundColorDefaulter.cs

```diff
@@ -0,0 +1,290 @@
+                    {
+                        var schemeThemeColors = scheme[themeId];
+
+                        if (AreClassificationsDefaultableToScheme(themeId, schemeThemeColors))
+                        {
+                            return true;
+                        }
+                    }
+                }
+                finally
+                {
+                    _fontAndColorStorage.CloseCategory();
+                }
+
+                return false;
+            }
+
+            private bool IsSupportedTheme(Guid themeId)
+            {
+                return _colorSchemes.Any(scheme => scheme.ContainsKey(themeId));
+            }
+
+            private bool AreClassificationsDefaultableToScheme(Guid themeId, ImmutableDictionary<string, uint?> schemeThemeColors)
+            {
+                foreach (var classification in Classifications)
+                {
+                    var colorItems = new ColorableItemInfo[1];
+
+                    if (_fontAndColorStorage.GetItem(classification, colorItems) != VSConstants.S_OK)
+                    {
+                        // Classifications that are still defaulted will not have entries.
+                        continue;
+                    }
+
+                    var colorItem = colorItems[0];
+
+                    if (!IsClassificationDefaultable(themeId, schemeThemeColors, colorItem, classification))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+
+            private bool IsClassificationDefaultable(Guid themeId, ImmutableDictionary<string, uint?> schemeThemeColors, ColorableItemInfo colorItem, string classification)
+            {
+                if (_fontAndColorUtilities.GetColorType(colorItem.crForeground, out var foregroundColorType) != VSConstants.S_OK)
+                {
+                    // Without being able to check color type, we cannot make a determination.
+                    return false;
+                }
+
+                return foregroundColorType switch
+                {
+                    (int)__VSCOLORTYPE.CT_AUTOMATIC => true,
+                    (int)__VSCOLORTYPE.CT_RAW => CanItemBeDefaulted(themeId, schemeThemeColors, colorItem, classification),
+                    _ => false
+                };
+            }
+
+            /// <summary>
+            /// Determines if the ColorableItemInfo can be reverted to its default state. This requires checking both color and font configuration,
+            /// since reverting will reset all information for the item.
+            /// </summary>
+            private bool CanItemBeDefaulted(Guid themeId, ImmutableDictionary<string, uint?> schemeThemeColors, ColorableItemInfo colorItem, string classification)
+            {
+                if (_fontAndColorUtilities.GetColorType(colorItem.crBackground, out var backgroundColorType) != VSConstants.S_OK)
```

Comment:
When we `RevertItemToDefault` it will default the Foreground, Background, and FontFlags. If a user had customized their background, I wanted to preserve it.